### PR TITLE
ci: make executor registry configurable

### PR DESCRIPTION
**Pull request by burmudar**
While working on a POC for pushing images to different registries I encountered this and thought it would be a good thing to have for other future changes.

This makes it so that we can switch the registries for (a) executors images are pulled from (b) the registry the server executors will use during test
## Test plan
CI

### CODE REVIEW

File: dev/ci/internal/ci/images_operations.go [hunk 1]
```diff
@@ -87,19 +87,18 @@ func bazelPushImagesCmd(c Config, isCandidate bool, opts ...bk.StepOpt) func(*bk
...
 		candidate = "true"
 	}
+	// Default registries.
+	devRegistry := images.SourcegraphDockerDevRegistry
+	prodRegistry := images.SourcegraphDockerPublishRegistry
 
 	// If we're building an internal release, we push the final images to that specific registry instead.
 	// See also: release_operations.go
-	if c.RunType.Is(runtype.InternalRelease) {
+	if c.RunType == runtype.InternalRelease {
```
Comment (burmudar) [inline dev/ci/internal/ci/images_operations.go:101]:
```suggestion
	if c.RunType.Is(runtype.InternalRelease) {
```

**Comment by jhchabran** [comment]
@burmudar I think you most likely want to get @BolajiOlajide in the loop here :)

**Review by jhchabran — APPROVED** [review]

**Comment by burmudar** [comment]
> @burmudar I think you most likely want to get @BolajiOlajide in the loop here :)

@jhchabran whoops - automerge was on. @BolajiOlajide can you take a look here and see if you approve?

**Review by BolajiOlajide — COMMENTED** [review]
LGTM

**Pull request merged**